parse/parse.d.ts

```diff
@@ -319,7 +322,9 @@ declare namespace Parse {
         static fetchAll<T>(list: Object[], options: SuccessFailureOptions): Promise<T>;
         static fetchAllIfNeeded<T>(list: Object[], options: SuccessFailureOptions): Promise<T>;
         static destroyAll<T>(list: Object[], options?: Object.DestroyAllOptions): Promise<T>;
-        static saveAll<T>(list: Object[], options?: Object.SaveAllOptions): Promise<T>;
+        static saveAll<T extends Object>(list: T[], options?: Object.SaveAllOptions): Promise<T[]>;
+
+        static registerSubclass<T extends Object>(className: string, clazz: new (options?: any) => T): void;
```

Comment:
All correct: I did not want to use a key word as parameter name (can you even do that in typescript?), so I used a classical Java tactic and named it clazz.